src/Provider/Vkontakte.php

```diff
@@ -1,51 +1,78 @@
 <?php
-/*!
-* Hybridauth
-* https://hybridauth.github.io | https://github.com/hybridauth/hybridauth
-*  (c) 2017 Hybridauth authors | https://hybridauth.github.io/license.html
-*/
 
 namespace Hybridauth\Provider;
 
 use Hybridauth\Adapter\OAuth2;
 use Hybridauth\Exception\UnexpectedApiResponseException;
-use Hybridauth\Data;
-use Hybridauth\User;
+use Hybridauth\Data\Collection;
+use Hybridauth\User\Profile;
 
 /**
- * Vkontakte OAuth2 provider adapter.
+ * Vkontakte provider adapter.
+ *
+ * Example:
+ *
+ *   $config = [
+ *       'callback'  => Hybridauth\HttpClient\Util::getCurrentUrl(),
+ *       'keys'      => ['id' => '', 'secret' => ''],
+ *   ];
+ *
+ *   $adapter = new Hybridauth\Provider\Vkontakte($config);
+ *
+ *   try {
+ *       if (!$adapter->isConnected()) {
+ *           $adapter->authenticate();
+ *       }
+ *
+ *       $userProfile = $adapter->getUserProfile();
+ *   }
+ *   catch(\Exception $e) {
+ *       print $e->getMessage() ;
+ *   }
  */
 class Vkontakte extends OAuth2
 {
     /**
-    * {@inheritdoc}
-    */
+     * {@inheritdoc}
+     */
     protected $apiBaseUrl = 'https://api.vk.com/method/';
 
     /**
-    * {@inheritdoc}
-    */
-    protected $authorizeUrl = 'http://api.vk.com/oauth/authorize';
+     * {@inheritdoc}
+     */
+    protected $authorizeUrl = 'https://api.vk.com/oauth/authorize';
 
     /**
-    * {@inheritdoc}
-    */
+     * {@inheritdoc}
+     */
     protected $accessTokenUrl = 'https://api.vk.com/oauth/token';
 
     /**
-    * {@inheritdoc}
-    */
-    protected $scope = 'email';
+     * {@inheritdoc}
+     */
+    protected $scope = 'email,offline';
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function hasAccessTokenExpired()
+	{
+		// As we using offline scope, $expired will be false.
+		$expired = $this->getStoredData('expires_in')
```

Comment:
We can't to validate by 'expires_at', because it's calculated value and it always will be less than current time at expires_in equals 0 (if pass 'offline' in scope, expires_in will be 0).